### DISCUSSION TITLE
Zwei oder mehr Victron MPPT

### DISCUSSION THREAD

**Discussion by Sklapp87** [hoylabs/OpenDTU-OnBattery#727] [Q&A]
Hi Community,
kann mir jemand kurz erklären, welche Rolle die Ve Direct Schnittstelle im Zusammenspiel mit Solar Passthrough spielt ?
Ich möchte zwei US5000, zwei Victron MPPT Einsätzen. Das sollte doch auch funktionieren ohne die VE Direct Schnittstelle des zweiten MPPT abzufragen ?

**Comment by Manos1966** [comment]
Hallo @Sklapp87 

https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Dynamic-Power-Limiter-%E2%80%90-Solar-Passthrough

Solar Passthrough is a feature that, if activated:
- ensures that "any available solar power" is first used to cover the needs of the household. 
- Only surplus solar power will charge the battery.

Without Ve Direct Schnittstelle, "any available solar power" ist unbekannt. 
OpenDTU weiss nicht wieviel Solar Strom ist verfuegbar, also alles haengt davon ab, ob die (OpenDTU) Einstellungen der Batterie erlauben, dass der Inverter die Batterie benutzten darf um Strom zu produzieren.
Erlauben die OpenDTU Batterie Einstellungen dass die Batterie nicht benutzt werden darf (Batterie nicht genug geladen, Batterie darf nur Nachts benutz werden, usw usw), bleibt der Wechslrichter ausgeschaltet OBWOHL deine zwei MPPTs gerade Solar Strom produzieren.

**Reply by Sklapp87** [reply]
Danke für die Antwort. Wie kommt die OpenDTU denn an die Daten des zweiten MPPT wenn ich dafür Victron2MQTT nutze ?

**Reply by spcqike** [reply]
Bisher gar nicht. 

Welche zwei Victron sind / werden es denn? Könntest du auch einfach einen nehmen?

**Reply by Sklapp87** [reply]
Ja ich kann in der aktuellen Ausbaustufe auch einen MPPT nehmen aber lieber wären mir zwei. Es gibt doch hier in der Community ein bespiel Projekt, wo der zweite MPPT per MQTT angebunden ist.

**Reply by madmartin** [reply]
Mein Setup hat auch zwei Victron MPPT, aber nur einer ist an der OpenDTU angeschlossen und wird für das "solar passthrough" genutzt. Die Energie, die der zweite Victron MPPT liefert, wird so immer für die Batterieladung benutzt. Es funktioniert also problemlos, aber das "solar passthrough" ist dann nicht perfekt.

**Reply by rs3hc** [reply]
Da scheint aber was in Arbeit zu sein.

https://github.com/helgeerbe/OpenDTU-OnBattery/pull/706

Ich selbst nutze OpenDTU noch nicht.

Aber ich habe in meiner Firma 2 100/20 die mit folgendem Script und einem ESP8266 ausgelesen werden.

https://github.com/KinDR007/VictronMPPT-ESPHOME

Technisch ist es also definitiv umsetzbar

**Reply by stanpete78** [reply]
I still believe a good solution would be if there were an option to just get the solar input value from a specified MQTT topic, so that no matter how many devices you have, you could just add them together and have the total amount. Even without a physical/serial connection between opendtu-onbattery and "all" the MPPTs.

**Reply by stanpete78** [reply]
I hope that a second MPPT will be supported sooner rather than later, to be honest. By the way, I'm running two MPPTs in the weirdest way possible: Each has their own ESP32 and pushes the VE Direct data to a MQTT server. Those two values are read and added with Homeassistant. In Home Assistant I have an automation that pushes a JSON payload with the current solar power value via curl to openDTU. This is slow and wearing down the EEPROM, but ... it works :-)

**Reply by schlimmchen** [reply]
> I hope that a second MPPT will be supported sooner rather than later, to be honest.

Shall I start a gofundme campain for @Gumbagubanga and me to expedite this?

> In Home Assistant I have an automation that pushes a JSON payload with the current solar power value via curl to openDTU

What value are you writing?

> wearing down the EEPROM

NAND flash, but nevermind :stuck_out_tongue:

**Reply by stanpete78** [reply]
I could spare a dime if you set up a gofundme :)

I‘m sending the entire batch of powerlimiter values but I‘m just changing the upperpowerlimit to the current max solar input every 15 minutes if the battery is below a certain threshhold (I think, I set it to 50%). This reduces the wear on the battery, but I might have to buy a new ESP once a year, if I like to stay within the promised NAN:D write cycles.

**Reply by stanpete78** [reply]
@schlimmchen  Here they call it EEPROM, by the way.
https://en.giantec-semi.com/blog/How-many-times-can-you-write-to-ESP32-EEPROM-Explore-the-secrets-of-chip-life#:~:text=The%20official%20guide%20mentions%20that,EEPROM%20when%20designing%20their%20applications.

**Reply by schlimmchen** [reply]
https://www.espressif.com/sites/default/files/documentation/esp32_datasheet_en.pdf
https://www.espressif.com/sites/default/files/documentation/esp32-s3_datasheet_en.pdf

The ESP32 packages have no in-package EEPROM. And at least in my vocabulary, EEPROM and NAND flash are not comparable but completely different technologies.

**Reply by stanpete78** [reply]
You are right. It seems like the ESP is sort of emulating an EEPROM but is actually writing to flash. That‘s why some call it EEPROM, incorrectly.

**Reply by stanpete78** [reply]
@schlimmchen  Just to add this. My strange way of automating with Homeassistant also works without additional ESPs and sending their Ve.direct output to an MQTT server.
 If you calculate your total solar power using the battery power values and the inverter output you get an OK estimate. If you run it through a filter, you get a very realiable estimate of how much solar power in total is being delivered into your system. I know this is all not a final solution, but I've been happily running this for a while now.
 
<img width="424" alt="Bildschirmfoto 2024-03-13 um 09 37 50" src="https://github.com/helgeerbe/OpenDTU-OnBattery/assets/73825064/43f0ddad-bd2a-44cd-b83c-104035326c90">

**Comment by Sklapp87** [comment]
@schlimmchen what is the Progress of the implemention of two mppt ?

**Reply by schlimmchen** [reply]
We will get there. Be patient, stop nagging. I know this is a popular request. As are many others. Count on me to help get this into a release, as I want this, too. Not today. Not tomorrow. By the end of the weekend might be feasible. PR #706 is good, but not quite ready and needs testing.

**Reply by Gumbagubanga** [reply]
I just finished a 3 days training and will push some code updates in the next couple days.
As @schlimmchen said: We'll get there sooner than later :)

Thank you all for your patience and anticipation 👍

**Comment by DerHavey1994** [comment]
@Sklapp87 Hier ein Link wie ich es gemacht habe: **https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/360#discussioncomment-7973979** 

Wenn alle Module dieselbe Ausrichtung haben und an beiden MPPT´s ca dasselbe ankommt, dann kannst du so wie ich es gemacht habe einfach die Leistung in der Firmware verdoppeln und du kommst ansatzweise und mit wenig Aufwand an den richtigen Solar Passthrough (Vielen Dank nochmal an @dirk-schulz für den Tipp).

Ein Victron muss dafür natürlich an OpenDTU angeschlossen sein.

Ich habe es zuletzt mit der Version 2023.12.27.post1 gemacht, ob sich zwischendurch was am Code geändert hat kann ich nicht sagen.

**Comment by Gumbagubanga** [comment]
Hi @Sklapp87 

Mitlerweile ist das Release mit dem Support für einen zweiten Victron MPPT raus.
Hierfür muss analog zum ersten Tracker ein Input-tauglicher Pin des ESP32 mit dem Victron MPPT Tracker (idealerweise über einen Opto-Koppler, siehe Doku) verbunden und in Pin-Mapping zusätzlich der `rx2` Pin definiert werden.

In https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/797 wurden bereits einige Punkte besprochen.
